tests/unit/conftest.py

```diff
@@ -118,11 +120,30 @@ def test_dir(tmpdir_factory) -> str:
     return str(tmp_path)
 
 
-@pytest.fixture(scope="session")
-def config(test_dir) -> Config:
-    """A good/known Config object"""
-    return Config(
-        f"{test_dir}/pipeline.yaml",
-        f"{test_dir}/input_data.yaml",
-        f"{test_dir}/environment.yaml",
+@pytest.fixture()
```

Comment:
This existing Config() fixture is a bit problematic now, because if we don't mock out the implementation validation, it fails, since we moved up all the validations. I wanted to avoid extraneous mocking of the same thing over and over in test_config, but I ended up having to duplicate some code in other places.

In general, it would be good to either / both:
a) add the right implementation metadata used here to the `tmp_dir` so we don't need to mock it at all
b) modularize  the way we mock out different bits and pieces; I tried to figure out how to do this with decorators but I had a hard time figuring it out.